### PR TITLE
Promote PodTopologySpread to stable in 1.19

### PR DESCRIPTION
**Pull request by Huang-Wei**
/sig scheduling

### CODE REVIEW

**Comment by Huang-Wei** [comment]
@sftim Thanks for the reminder. This PR is ready for review.

I've rebased with dev-1.19. IIRC it's not necessary to manually update the featuregate in `content/en/docs/reference/command-line-tools-reference/{component}.md`, as there is an automation script doing that, right? If so, in this PR, I just updated `content/en/docs/reference/command-line-tools-reference/feature-gates.md` manually.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ different Kubernetes components.
 | `ExpandPersistentVolumes` | `true` | Beta | 1.11 | |
 | `ExperimentalHostUserNamespaceDefaulting` | `false` | Beta | 1.5 | |
 | `EvenPodsSpread` | `false` | Alpha | 1.16 | 1.17 |
-| `EvenPodsSpread` | `true` | Beta | 1.18 | |
+| `EvenPodsSpread` | `true` | Beta | 1.18 | 1.18 |
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:103]: I'm afraid you need to do this differently. Remove the feature gate from _Feature gates for Alpha or Beta features_ and move it to the table for stable and graduated features.

Comment (Huang-Wei) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:103]: Ah, definitely. Updated.

**Comment by celestehorgan** [comment]
/label needs-rebase

@Huang-Wei could you rebase this please?

**Comment by Huang-Wei** [comment]
@celestehorgan sure, rebase completed.

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 2]
```diff
@@ -265,13 +256,11 @@ scheduled - more packed or more scattered.
 - For `PodAntiAffinity`, only one Pod can be scheduled into a
   single topology domain.
 
-The "EvenPodsSpread" feature provides flexible options to distribute Pods evenly across different
+The "PodTopologySpread" feature provides flexible options to distribute Pods evenly across different
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:259]: You don't need to name feature gates for graduated features.

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 3]
```diff
@@ -265,13 +256,11 @@ scheduled - more packed or more scattered.
...
 topology domains - to achieve high availability or cost-saving. This can also help on rolling update
 workloads and scaling out replicas smoothly. See [Motivation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/895-pod-topology-spread#motivation) for more details.
 
 ## Known Limitations
 
 - Scaling down a Deployment may result in imbalanced Pods distribution.
 - Pods matched on tainted nodes are respected. See [Issue 80921](https://github.com/kubernetes/kubernetes/issues/80921)
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:266]: Is this line still relevant?

**Comment by sftim** [comment]
@kubernetes/sig-scheduling-pr-reviews please take a look - if happy, add /lgtm

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md (hunk 3 above)
Comment (Huang-Wei) [reply content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:266]: Yes.

**Comment by Huang-Wei** [comment]
@sftim Thanks for the review. Comments addressed. PTAL.

**Comment by annajung** [comment]
Hi 👋 1.19 docs shadow here,
Can we get a tech review from SIG Scheduling? @kubernetes/sig-scheduling-pr-reviews

**Comment by Huang-Wei** [comment]
@alculquicondor Can you please take another look, if possible /lgtm?

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 4]
```diff
@@ -265,13 +256,14 @@ scheduled - more packed or more scattered.
 - For `PodAntiAffinity`, only one Pod can be scheduled into a
   single topology domain.
 
-The "EvenPodsSpread" feature provides flexible options to distribute Pods evenly across different
-topology domains - to achieve high availability or cost-saving. This can also help on rolling update
-workloads and scaling out replicas smoothly. See [Motivation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/895-pod-topology-spread#motivation) for more details.
+To fill the gap, you can specify topology spread constraints to distribute Pods
+across different topology domains flexibly - to achieve either high availability
```
Comment (alculquicondor) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:260]:
```suggestion
For finer control, you can specify topology spread constraints to distribute Pods
across different topology domains - to achieve either high availability
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 5]
```diff
@@ -224,6 +222,9 @@ different Kubernetes components.
 | `EnableEquivalenceClassCache` | - | Deprecated | 1.15 | - |
 | `ExperimentalCriticalPodAnnotation` | `false` | Alpha | 1.5 | 1.12 |
 | `ExperimentalCriticalPodAnnotation` | `false` | Deprecated | 1.13 | - |
+| `EvenPodsSpread` | `false` | Alpha | 1.16 | 1.17 |
+| `EvenPodsSpread` | `true` | Beta | 1.18 | 1.18 |
+| `EvenPodsSpread` | `true` | GA | 1.19 | - |
```
Comment (alculquicondor) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:227]: Isn't the gate removed in 1.19?

**Comment by savitharaghunathan** [comment]
Hi @Huang-Wei 👋 , Please rebase to fix the merge conflict.

**Comment by Huang-Wei** [comment]
Github seems to have a problem in showing up the latest code changes of a branch. For instance, my [forked repo](https://github.com/Huang-Wei/website/commits/1.19-pts-stable) has the latest change of this PR, but not showing up here.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 5 above)
Comment (Huang-Wei) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:228]: I think it serves as listing historical changes.

Comment (alculquicondor) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:228]: @sftim for authoritative answer, but the historical purpose should be satisfied with the Alpha and Beta rows.

**Comment by Huang-Wei** [comment]
> Almost forgot, could you update the interpretation of whenUnschedulable and maxSkwe as we updated it in the API docs.

@alculquicondor PTAL. Also I added a link to the blog in the end.

**Comment by Huang-Wei** [comment]
Kindly ping @sftim for a final look.

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 6]
```diff
@@ -16,13 +14,6 @@ You can use _topology spread constraints_ to control how {{< glossary_tooltip te
 
 ## Prerequisites
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:15]: @Huang-Wei ,
Are there prerequisites?

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 5 above)
Comment (kbhawkey) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md:228]: This looks correct. Once a feature reaches GA, move the alpha+beta items to `stable/deprecated` table and add `GA or deprecated`.

**Comment by kbhawkey** [comment]
Previews:
https://deploy-preview-21115--kubernetes-io-vnext-staging.netlify.app/docs/reference/command-line-tools-reference/feature-gates/

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md (hunk 6 above)
Comment (kbhawkey) [reply content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:15]: Could you rename this heading?

**Comment by kbhawkey** [comment]
@Huang-Wei , looks fine. Just questioned the `Prerequisites` heading.
/lgtm

**Comment by savitharaghunathan** [comment]
@kbhawkey Is it good to merge or the question regarding Prerequisites needs to be addressed before merging?

**Comment by kbhawkey** [comment]
@alculquicondor , Would you provide a technical LGTM? Thanks!

**Comment by kbhawkey** [comment]
It can merge as is.

**Comment by alculquicondor** [comment]
I would put the feature gate as deprecated, given that it was removed.

**Comment by kbhawkey** [comment]
> I would put the feature gate as deprecated, given that it was removed.

Checking: EvenPodsSpread is the feature that has reached GA in 1.19?
A feature reaches GA, not the feature gate.
```
If a feature is stable you can find all stages for that feature listed in the
  [Graduated/Deprecated feature gate table](#feature-gates-for-graduated-or-deprecated-features).
```
It is somewhat confusing as the blog post mentions `PodTopologySpread`.

**Comment by alculquicondor** [comment]
Oh I see. I was misinterpreting the role of this list.

We renamed the underlying scheduling plugin name. But the feature name
remained as originally.

/lgtm

**Comment by Huang-Wei** [comment]
@kbhawkey The feature gate was named `EvenPodsSpread` since alpha, however, we realized that technically the word "PodTopologySpread" is more proper.

**Comment by kbhawkey** [comment]
Great! @savitharaghunathan , this is ready to merge. Thanks.

File: content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md [hunk 7]
```diff
@@ -70,7 +61,15 @@ spec:
...
 
-- **maxSkew** describes the degree to which Pods may be unevenly distributed. It's the maximum permitted difference between the number of matching Pods in any two topology domains of a given topology type. It must be greater than zero.
+- **maxSkew** describes the degree to which Pods may be unevenly distributed.
+  It's the maximum permitted difference between the number of matching Pods in
+  any two topology domains of a given topology type. It must be greater than
+  zero. Its semantics differs according to the value of `whenUnsatisfiable`:
+  - when `whenUnsatisfiable` equals to "DoNotSchedule", `maxSkew` is the maximum
+    permitted difference between the number of matching pods in the target
+    topology and the global minimum.
+  - when `whenUnsatisfiable` equals to "ScheduleAnyway", scheduler gives higher
```
Comment (sftim) [inline content/en/docs/concepts/workloads/pods/pod-topology-spread-constraints.md:71]: _nit (OK to skip / fix after merge):_
“is equal to” or “equals”, not “equals to” (that last phrasing is not idiomatic English)

**Pull request merged**